packages/flutter/lib/src/material/desktop_text_selection_toolbar.dart

```diff
@@ -0,0 +1,88 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'material.dart';
+import 'text_selection_toolbar.dart';
+
+const double _kToolbarScreenPadding = 8.0;
+const double _kToolbarWidth = 222.0;
+
+/// A Material-style desktop text selection toolbar.
+///
+/// Typically displays buttons for text manipulation, e.g. copying and pasting
+/// text.
+///
+/// Tries to position itself as closely as possible to [anchor] while remaining
+/// fully on-screen.
+///
+/// See also:
+///
+///  * [DefaultTextSelectionToolbar], which builds the toolbar for the current
+///    platform.
+///  * [TextSelectionToolbar], which is similar, but builds an Android-style
+///    toolbar.
+class DesktopTextSelectionToolbar extends StatelessWidget {
+  /// Creates an instance of _DesktopTextSelectionToolbar.
+  const DesktopTextSelectionToolbar({
+    super.key,
+    required this.anchor,
+    required this.children,
+  }) : assert(children.length > 0);
+
+  /// {@template flutter.material.DesktopTextSelectionToolbar.anchor}
+  /// The point at which the toolbar will attempt to position itself as closely
+  /// as possible.
+  /// {@endtemplate}
+  final Offset anchor;
+
+  /// {@macro flutter.material.TextSelectionToolbar.children}
+  ///
+  /// See also:
+  ///   * [DesktopTextSelectionToolbarButton], which builds a default
+  ///     Material-style desktop text selection toolbar text button.
+  final List<Widget> children;
+
+  // Builds a desktop toolbar in the Material style.
+  static Widget _defaultToolbarBuilder(BuildContext context, Widget child) {
+    return SizedBox(
+      width: _kToolbarWidth,
+      child: Material(
+        borderRadius: const BorderRadius.all(Radius.circular(7.0)),
+        clipBehavior: Clip.antiAlias,
+        elevation: 1.0,
+        type: MaterialType.card,
+        child: child,
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    assert(debugCheckHasMediaQuery(context));
+    final MediaQueryData mediaQuery = MediaQuery.of(context);
+
+    final double paddingAbove = mediaQuery.padding.top + _kToolbarScreenPadding;
+    final Offset localAdjustment = Offset(_kToolbarScreenPadding, paddingAbove);
+
+    return Padding(
+      padding: EdgeInsets.fromLTRB(
+        _kToolbarScreenPadding,
+        paddingAbove,
+        _kToolbarScreenPadding,
+        _kToolbarScreenPadding,
+      ),
+      child: CustomSingleChildLayout(
+        delegate: DesktopTextSelectionToolbarLayoutDelegate(
+          anchor: anchor - localAdjustment,
+        ),
+        child: _defaultToolbarBuilder(context, Column(
```

Comment:
Nit: trailing comma, and formatting: `context`, and `Column` on their own lines.